x/auth/signing/sign_mode_handler.go

```diff
@@ -0,0 +1,35 @@
+package signing
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+// SignModeHandler defines a interface to be implemented by types which will handle
+// SignMode's by generating sign bytes from a Tx and SignerData
+type SignModeHandler interface {
+	// DefaultMode is the default mode that is to be used with this handler if no
+	// other mode is specified. This can be useful for testing and CLI usage
+	DefaultMode() txtypes.SignMode
+
+	// Modes is the list of modes supporting by this handler
+	Modes() []txtypes.SignMode
+
+	// GetSignBytes returns the sign bytes for the provided SignMode, SignerData and Tx,
+	// or an error
+	GetSignBytes(mode txtypes.SignMode, data SignerData, tx sdk.Tx) ([]byte, error)
```

Comment:
This relates to my previous question. Being that this will operate on a concrete implementation, what's the point of passing the mode.

Reply:
A single handler can handle multiple modes, in particular `SignModeHandlerMap`

Reply:
Ahh right, because SignModeHandlerMap 👍